aibolit/metrics/cognitiveC/cognitive_c.py

```diff
@@ -2,36 +2,65 @@
 from aibolit.patterns.var_middle.var_middle import JavalangImproved
 from typing import List
 
+increment_for = [
+    javalang.tree.IfStatement,
+    javalang.tree.SwitchStatement,
+    javalang.tree.ForStatement,
+    javalang.tree.WhileStatement,
+    javalang.tree.DoStatement,
+    javalang.tree.CatchClause,
+    javalang.tree.BreakStatement,
+    javalang.tree.ContinueStatement,
+    javalang.tree.TernaryExpression,
+    # javalang.tree.BinaryOperation,
+]
+
+nested_for = [
+    javalang.tree.IfStatement,
+    javalang.tree.SwitchStatement,
+    javalang.tree.ForStatement,
+    javalang.tree.WhileStatement,
+    javalang.tree.DoStatement,
+    javalang.tree.CatchClause,
+]
+
 
 class CognitiveComplexity:
     '''
     beta version of Cognitive Complexity
     '''
     def __init__(self):
-        pass
+        self.complexity = 0
 
-    def if_statements(self, nodes: List) -> int:
-        count = 0
-        for i in nodes:
-            if type(i.node) in [javalang.tree.IfStatement]:
-                count += 1
-        return count
+    def traverse_childs(self, block: javalang.tree, nested_level: int):
+        for each_child in block.children:
+            self.get_complexity(each_child, nested_level)
 
-    def for_statements(self, nodes: List) -> int:
-        count = 0
-        for i in nodes:
-            if type(i.node) in [javalang.tree.ForStatement]:
-                count += 1
-        return count
+    def get_complexity(self, childs, nested_level: int):
+        child_arr = childs if isinstance(childs, List) else [childs]
 
-    def value(self, filename: str):
+        for c in child_arr:
 
-        nodes = JavalangImproved(filename).tree_to_nodes()
+            if hasattr(c, 'children'):
+                if type(c) in increment_for and type(c) in nested_for:
+                    self.complexity += 1 + nested_level
+                    self.traverse_childs(c, nested_level + 1)
 
-        rules = [self.if_statements, self.for_statements]
-        complexity = 0
+                elif type(c) in increment_for and type(c) not in nested_for:
+                    self.complexity += 1
+                    self.traverse_childs(c, nested_level)
+                else:
+                    self.traverse_childs(c, nested_level)
 
-        for rule in rules:
-            complexity += rule(nodes)
+            continue
+
+    def value(self, filename: str) -> int:
+
+        nodes = JavalangImproved(filename).tree_to_nodes()
```

Comment:
Do not use JavaLangImproved if you can use JavaLang pure

Reply:
fixed